src/nvim/cursor_shape.c

```diff
@@ -230,8 +230,9 @@ char_u *parse_shape_opt(int what)
         }
       }
       modep = p;
-      if (*modep == ',')
-        ++modep;
+      if (modep != NULL && *modep == ',') {
```

Comment:
I think this warning is caused by the `assert(modep < colonp)` on line 121 (added by me) which makes no sense given that the next line in `while (modep < colonp || all_idx >= 0)`. That assert was trying to silence a different error, but I think it should be removed and revisited.

Reply:
`p` is `NULL` here if there is no comma after the colon.